netty/src/main/java/io/grpc/netty/GrpcSslContexts.java

```diff
@@ -176,5 +187,15 @@ static void ensureAlpnAndH2Enabled(ApplicationProtocolNegotiator alpnNegotiator)
       throw new IllegalArgumentException("This ALPN config does not support HTTP/2. Expected '"
           + HTTP2_VERSION + "', but got " + alpnNegotiator.protocols() + '.');
     }
+    if (!alpnNegotiator.protocols().contains(GRPC_EXP_VERSION)) {
```

Comment:
The check here only applies to the ApplicationProtocolConfig we build up in GrpcSslContexts.java, and is a sanity check used when constructing a Netty{Channel,Server}Builder. So, it's not about wire level consideration of whether GRPC_EXP_VERSION is presented by the other party in a TLS negotiation, but instead a check on whether we're correctly plumbing this around the grpc-java Netty support.